chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/ShapeFilter.java

```diff
@@ -0,0 +1,219 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Lablicate GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Alexander Stark - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.chemclipse.xxd.model.filter.peaks;
+
+import java.util.Collection;
+import java.util.function.BiPredicate;
+import org.eclipse.chemclipse.model.core.IPeak;
+import org.eclipse.chemclipse.model.core.IPeakModel;
+import org.eclipse.chemclipse.model.filter.IPeakFilter;
+import org.eclipse.chemclipse.processing.Processor;
+import org.eclipse.chemclipse.processing.core.MessageConsumer;
+import org.eclipse.chemclipse.processing.filter.CRUDListener;
+import org.eclipse.chemclipse.processing.filter.Filter;
+import org.eclipse.chemclipse.xxd.model.support.PeakShapeFilterSelectionCriterion;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.SubMonitor;
+import org.osgi.service.component.annotations.Component;
+
+import com.google.common.collect.Range;
+
+@Component(service = { IPeakFilter.class, Filter.class, Processor.class })
+public class ShapeFilter implements IPeakFilter<ShapeFilterSettings> {
+
+	private static BiPredicate<Double, Double> LEADING_SMALLER_THAN_LIMIT_COMPARATOR = (value, shapeSetting) -> (value < shapeSetting);
+	private static BiPredicate<Double, Double> TAILING_GREATER_THAN_LIMIT_COMPARATOR = (value, shapeSetting) -> (value > shapeSetting);
+	private static BiPredicate<RangeContainer, Range<Double>> VALUES_WITHIN_RANGE_COMPARATOR = (container, shapeSetting) -> (shapeSetting.contains(container.leading) && (shapeSetting.contains(container.tailing)));
+
+	private static class ShapePredicate<T> {
```

Comment:
Try to use only one predicate, e.g.
MyPredicate<A,B>, where A and B can bei either Double or Range<Double>.